docs/GDPs/gdp-0003-duck-storage.rst

```diff
@@ -0,0 +1,750 @@
+    the respective dimension, which can be used to denote asymmetric halos. It defaults to no halo,
+    i.e. :code:`(0, 0, 0)`. (See also Section :ref:`domain_and_halo`)
+
+:code:`layout: Optional[Sequence[int]]`
+    A permutation of integers in :code:`[0 .. ndim-1]`. It indicates the order of strides in
+    decreasing order. I.e. "0" indicates that the stride in that dimension is the largest, while the
+    largest entry in the layout sequence corresponds to the dimension with the smallest stride, which
+    typically is contiguous in memory.
+
+    Default values as indicated by the :code:`defaults` parameter may depend on the dimensions. E.g.
+    if :code:`defaults` is any of the compiled GridTools backends, the default value is defined
+    according to the semantic meaning of each dimension. For example for the :code:`"gtx86"`
+    backend, the smallest stride is always in the K dimension, independently of which index
+    corresponds to the K dimension. On the other hand, we assume that if a storage is created from
+    an existing FORTRAN array, the first index has the smallest stride, irrespective of its
+    corresponding axis. I.e. the layout of a 3d storage is always :code:`(2, 1, 0)` for both IJK and
+    KJI storages.
+
+    .. list-table:: Default :code:`layout` parameter when given :code:`defaults` and :code:`dims`
+       :header-rows: 1
+       :stub-columns: 1
+
+       * -
+         - :code:`defaults="F"`
+         - :code:`defaults="C"`
+         - :code:`defaults="gtx86"`
+         - :code:`defaults="gtcuda"`
+
+       * - :code:`dims="IJK"`
+         - :code:`layout=(2, 1, 0)`
+         - :code:`layout=(0, 1, 2)`
+         - :code:`layout=(0, 1, 2)`
+         - :code:`layout=(2, 1, 0)`
+
+       * - :code:`dims="KJI"`
+         - :code:`layout=(2, 1, 0)`
+         - :code:`layout=(0, 1, 2)`
+         - :code:`layout=(2, 1, 0)`
+         - :code:`layout=(0, 1, 2)`
+
+    The rationale behind this is that in this way, storages allocated with :code:`defaults` set to a
+    backend will always get optimal performance, while :code:`defaults` set to :code:`"F"` or
+    :code:`"C"` will have expected behavior when wrapping FORTRAN or C buffers, respectively.
+
+:code:`managed: Optional[str]`
+    :code:`None`, :code:`"gt4py"` or :code:`"cuda"`. It only has effect if :code:`device="gpu"` and
+    it specifies whether the synchronization between the host and device buffers is not done
+    (:code:`None`), GT4Py (:code:`"gt4py"`) or CUDA (:code:`"cuda"`). It defaults to :code:`"gt4py"`
+
+The values of parameters which are not explicitly defined by the user will be inferred from the
+first alternative source where the parameter is defined in the following search order:
+
+1. The provided :code:`defaults` parameter set.
+2. The provided :code:`data` or :code:`device_data` parameters.
+3. A fallback default value specified above. The only case where this is not available is
+   :code:`shape`, in which case an exception is raised.
```

Comment:
This part about inferring parameters should probably be written at the start of this section. Which parameters can be derived from `data`/`device_data`, and how are they derived?